src/scripts/migrations/createBackendUsers.js

```diff
@@ -0,0 +1,549 @@
+/*
+  A script that loads all documents in db and create user for all occurrences of non-web users.
+
+  The followings are current references to user records in db:
+    - analytics: (~1825k entries as of 2020/10)
+      - (docUserId, docAppId) via docId (reply/article)
+
+    - articlecategoryfeedbacks: (less than 1k  entries as of 2020/10)
+      id = ${articleId}__${categoryId}__${userId}__${appId},
+      - (userId, appId)
+
+    - articlereplyfeedbacks: (~179k entries as of 2020/10)
+      id = ${articleId}__${replyId}__${userId}__${appId},
+      - (userId, appId)
+
+    - articles:  (~ 40k entries as of 2020/10)
+      - (userId, appId)
+      - [references]: 
+          - (userId, appId)
+      - [articleReplies]:
+          - (userId, appId)
+      - [articleCategories]:
+          - (userId, appId)
+
+    - replies: (~42k entries as of 2020/10)
+      - (userId, appId)
+
+    - replyrequests: (~61k entries as of 2020/10)
+      id = ${articleId}__${userId}__${appId}
+      - (userId, appId)
+      - [feedbacks]:
+        - (userId, appId)
+*/
+
+import 'dotenv/config';
+import client from 'util/client';
+import Bulk from 'util/bulk';
+import rollbar from 'rollbarInstance';
+import {
+  generatePseudonym,
+  generateOpenPeepsAvatar,
+  AvatarTypes,
+  convertAppUserIdToUserId,
+} from 'graphql/models/User';
+import { get, set } from 'lodash';
+
+const AGG_NAME = 'userIdPair';
+
+const AGG_SCRIPT = `
+  boolean valueExists(def map, def key) { map.containsKey(key) && map[key] != ''}
```

Comment:
I am not familiar with painless. Can painless omit `return` ?

Reply:
https://www.elastic.co/guide/en/elasticsearch/painless/6.2/painless-syntax.html
It's not really well documented but from the look of the example and the fact it actually runs as expected, I think it does if the return type is not void.